src/core/modules/AtmosphereShowMySky.cpp

```diff
@@ -772,13 +838,12 @@ void AtmosphereShowMySky::computeColor(StelCore* core, const double JD, const Pl
 		return;
 	}
 
-
 	// FIXME: ignoring the "additional luminance" like star background etc.; see AtmospherePreetham for all potentially needed terms
 	const auto numViewRayGridPoints=(1+gridMaxX)*(1+gridMaxY);
 	for (int i=0; i<numViewRayGridPoints; ++i)
 	{
 		Vec3d point(1, 0, 0);
-		prj->unProject(posGrid[i][0],posGrid[i][1],point);
+		prj->unProject(posGrid[i][0]*(double)ppxatmo,posGrid[i][1]*(double)ppxatmo,point);
```

Comment:
This is imprecise, because `posGrid` points were generated assuming width&height divided by `ppxatmo` **and rounded** down. Multiplying back we won't get the full width&height.
Instead, we need to multiply these two coordinates by `double(fullWidth)/reducedWidth` and `double(fullHeight)/reducedHeight` respectively.

Reply:
You are absolutely right!